perfkitbenchmarker/providers/openstack/os_virtual_machine.py

```diff
@@ -54,21 +55,47 @@ def __init__(self, vm_spec):
                                                FLAGS.run_uri)
         self.client = os_utils.NovaClient()
         self.public_network = os_network.OpenStackPublicNetwork(
-            FLAGS.openstack_public_network
+            FLAGS.openstack_floating_ip_pool
         )
         self.id = None
         self.pk = None
         self.user_name = FLAGS.openstack_image_username
         self.boot_wait_time = None
         self.image = self.image or self.DEFAULT_IMAGE
+        self.public_net = None
+        self.private_net = None
+        self.floating_ip = None
 
     def _Create(self):
         image = self.client.images.findall(name=self.image)[0]
         flavor = self.client.flavors.findall(name=self.machine_type)[0]
 
-        network = self.client.networks.find(
-            label=FLAGS.openstack_private_network)
-        nics = [{'net-id': network.id}]
+        # FIXME(meteofox): Remove --openstack_public_network and
+        # --openstack_private_network once depreciation time has expired
+        self.floating_ip_pool_name = (FLAGS.openstack_floating_ip_pool or
+                                      FLAGS.openstack_public_network)
+        self.network_name = (FLAGS.openstack_network or
+                             FLAGS.openstack_private_network)
+
+        if self.network_name:
+            self.private_net = self.client.networks.find(
+                label=self.network_name)
+        if self.floating_ip_pool_name:
+            self.public_net = self.client.networks.find(
+                label=self.floating_ip_pool_name)
+
+        if self.public_net is None and self.private_net is None:
+            raise errors.Error(
+                'Cannot build instance without a network. Make sure to set '
+                'either just --openstack_network or both --openstack_network '
+                'and --openstack_private_network flags.')
```

Comment:
Why both `--openstack_network` and `--openstack_private_network`? It looks like if both are set, then only `--openstack_network` is used.